### DISCUSSION TITLE
Upgrade sub-dependancies (yarn)

### DISCUSSION THREAD

**Discussion by fhp** [renovatebot/renovate#13772] [Request Help] [answered]
### What would you like Renovate to be able to do?

Renovatebot is awesome when it comes to upgrading direct dependencies, but  sub-dependencies are not upgraded, which can lead to failing builds due to version mismatches.

A very concrete example: `node-fetch` was upgraded to version `2.6.7`, producing the following `yarn.lock`.
```
node-fetch@2.6.7:
  version "2.6.7"
  resolved "https://registry.yarnpkg.com/node-fetch/-/node-fetch-2.6.7.tgz#24de9fba827e3b4ae44dc8b20256a379160052ad"
  integrity sha512-ZjMPFEfVx5j+y2yF35Kzx5sF7kDzxuDj6ziH4FFbOp87zKDZNx8yExJIb05OGF4Nlt9IHFIMBkRl41VdvcNdbQ==
  dependencies:
    whatwg-url "^5.0.0"

node-fetch@^2.3.0, node-fetch@^2.6.1:
  version "2.6.1"
  resolved "https://registry.yarnpkg.com/node-fetch/-/node-fetch-2.6.1.tgz#045bd323631f76ed2e2b55573394416b639a0052"
  integrity sha512-V4aYg89jEoVRxRb2fJdAg8FHvI7cEyYdVAh94HH0UIK8oJxUfkjlDQN9RbMx+bEjP7+ggMiFRprSti032Oipxw==
```

However, the older versions were not pinned to a specific version, so they could also be upgraded to `2.6.7`. So I would like them to be upgraded like this:

```
node-fetch@2.6.7, node-fetch@^2.3.0, node-fetch@^2.6.1:
  version "2.6.7"
  resolved "https://registry.yarnpkg.com/node-fetch/-/node-fetch-2.6.7.tgz#24de9fba827e3b4ae44dc8b20256a379160052ad"
  integrity sha512-ZjMPFEfVx5j+y2yF35Kzx5sF7kDzxuDj6ziH4FFbOp87zKDZNx8yExJIb05OGF4Nlt9IHFIMBkRl41VdvcNdbQ==
  dependencies:
    whatwg-url "^5.0.0"
```

I was unable to find a flag or setting to trigger this behavior, but if I missed it somehow, please name the setting and excuse me for the noise.

### If you have any ideas on how this should be implemented, please tell us here.

I did not look into the code, but with some pointers, I am willing to investigate further and create a PR.

### Is this a feature you are interested in implementing yourself?

Maybe

**Comment by viceice** [comment, marked as answer]
This is default yarn install behavior, you either need to configure `postUpdateOptions` or `lockFileMaintenance`.

- https://docs.renovatebot.com/configuration-options/#postupdateoptions
- https://docs.renovatebot.com/configuration-options/#lockfilemaintenance

**Reply by fhp** [reply]
Thanks :grinning:, `postUpdateOptions: yarnDedupeHighest` was exactly what I was looking for.
